src/utils/hooks.ts

```diff
@@ -244,6 +220,37 @@ export function useCortexApi<T>(
   }, deps);
 }
 
+export function useCortexApiMemorized<T>(
+  f: (api: CortexApi) => Promise<T>,
+  deps: any[] = [],
+) {
+  const cortexApi = useApi(cortexApiRef);
+
+  return useAsyncMemorized(async () => {
+    return await f(cortexApi);
+  }, deps);
+}
+
+function useAsyncMemorized<T>(f: () => Promise<T>, deps: any[]) {
+  const [cache, setCache] = useState<Record<string, any>>({});
+  const cacheKey = useMemo(() => JSON.stringify(deps), [deps]);
+  const cachedResult = useMemo(() => cache[cacheKey], [cacheKey, cache]);
```

Comment:
It doesn't look very complex to access an array, so I guess it doesn't have to be memoized, right?